src/dialogs/components/dialog/dialogField.ts

```diff
@@ -0,0 +1,81 @@
+import {DialogFieldClass} from '../../interfaces/abstractDialogFieldClass';
+import * as _ from 'lodash';
+/**
+ * This component deals with an individual dialog field
+ *
+ * @extends miqStaticAssets.dialog.DialogFieldClass
+ * @memberof miqStaticAssets.dialog
+ * @ngdoc controller
+ * @name DialogFieldController
+ */
+
+export class DialogFieldController extends DialogFieldClass {
+     public service: any;
+     public dialogValue: any;
+     public dialogField: any;
+     public clonedDialogField: any;
+
+  /*@ngInject*/
+  constructor(private DialogData: any) {
+      super();
+  }
+  /**
+   * Sets up the dialog field with defaults
+   * @memberof DialogFieldController
+   * @function $onInit
+   */
+  public $onInit() {
+    this.service = this.DialogData;
+    this.clonedDialogField = _.cloneDeep(this.field);
+    this.dialogField = this.service.setupField(this.field);
+
+    this.changesHappened(); // doing this so that the field value gets bubbled up to the dialog component
+  }
+  /**
```

Comment:
Would be nice to get some newlines here, at least between functions, and between classes.